sql/src/main/java/org/apache/druid/sql/calcite/planner/querygen/DruidQueryGenerator.java

```diff
@@ -280,10 +281,19 @@ public DruidQuery buildQuery(boolean topLevel)
         );
       }
 
+
+      private SourceDesc getSource()
+      {
+        if (source != null) {
+          return source;
+        }
+        return source = realGetSource();
+      }
```

Comment:
Complete and total nit, but "positive logic" is often easier to read.  I.e. 

```
if (source == null) {
  source = realGetSource();
}
return source
```

Does the exact same thing and has less negation, which, I think makes it easier to read.  I had to do an extra double check and think about what it means to return the result of the reference setting operation and then convince myself that this was actually caching.  The positive logic version is more explicit in caching and returning the cached reference.

Reply:
perfectly makes sense; I must have been in some weird mental state writing it like that